Pilot/pilotCommands.py

```diff
@@ -247,29 +248,39 @@ def _installDIRACpy3(self):
     """ Install python3 version of DIRAC client
     """
 
-    # 1. Download DIRACOS
+    # 1. Get the DIRACOS installer
     # curl -O -L https://github.com/DIRACGrid/DIRACOS2/releases/latest/download/DIRACOS-Linux-$(uname -m).sh
     try:
       machine = os.uname().machine  # py3
     except AttributeError:
       machine = os.uname()[4]  # py2
 
-    # FIXME: we should have a (set of) different location(s), starting from CVMFS
-    if not retrieveUrlTimeout(
-        "https://github.com/DIRACGrid/DIRACOS2/releases/latest/download/DIRACOS-Linux-%s.sh" % machine,
-        "DIRACOS-Linux-%s.sh" % machine,
-        self.log
-    ):
-      self.exitWithError(1)
+    installerName = "DIRACOS-Linux-%s.sh" % machine
+
+    # try to find the installer in CVMFS first
+    installer = '/cvmfs/dirac.egi.eu/installSource/%s' % installerName
+    if not os.path.exists(installer):
+      # Try to copy it locally if it is just not yet in the local CVMFS cache
+      shutil.copyfile(installer, installerName)
```

Comment:
This would raise an exception if it's really not in CVMFS (or if CVMFS is not mounted). And actually:

py2:
```
In [2]: shutil.copyfile('pippo', 'pluto')
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-2-607d8ae84fc8> in <module>()
----> 1 shutil.copyfile('pippo', 'pluto')

/home/toffo/mambaforge/envs/python2/lib/python2.7/shutil.pyc in copyfile(src, dst)
     80                 raise SpecialFileError("`%s` is a named pipe" % fn)
     81 
---> 82     with open(src, 'rb') as fsrc:
     83         with open(dst, 'wb') as fdst:
     84             copyfileobj(fsrc, fdst)

IOError: [Errno 2] No such file or directory: 'pippo'
```
(raising `IOError`)

py3: 
```
In [2]: shutil.copyfile('pippo', 'pluto')
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-2-607d8ae84fc8> in <module>
----> 1 shutil.copyfile('pippo', 'pluto')

~/mambaforge/envs/dirac-development/lib/python3.9/shutil.py in copyfile(src, dst, follow_symlinks)
    262         os.symlink(os.readlink(src), dst)
    263     else:
--> 264         with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
    265             # macOS
    266             if _HAS_FCOPYFILE:

FileNotFoundError: [Errno 2] No such file or directory: 'pippo'
```
raising py3-only FileNotFoundError

So at the beginning of the file you should add something like:
```
try:
  FileNotFoundError  # py3 only
except NameError:
  FileNotFoundError = OSError
```

Reply:
Indeed. I have changed the logic to be more simple and straightforward